homeassistant/components/flume/config_flow.py

```diff
@@ -0,0 +1,102 @@
+"""Config flow for flume integration."""
+from functools import partial
+import logging
+
+from pyflume import FlumeDeviceList
+from requests.exceptions import RequestException
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import BASE_TOKEN_FILENAME, CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+# If flume ever implements a login page for oauth
+# we can use the oauth2 support built into Home Assistant.
+#
+# Currently they only implement the token endpoint
+#
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_CLIENT_ID): str,
+        vol.Required(CONF_CLIENT_SECRET): str,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    username = data[CONF_USERNAME]
+    password = data[CONF_PASSWORD]
+    client_id = data[CONF_CLIENT_ID]
+    client_secret = data[CONF_CLIENT_SECRET]
+    flume_token_full_path = hass.config.path(f"{BASE_TOKEN_FILENAME}-{username}")
+    try:
+        flume_devices = await hass.async_add_executor_job(
+            partial(
+                FlumeDeviceList,
+                username,
+                password,
+                client_id,
+                client_secret,
+                flume_token_full_path,
+            )
+        )
```

Comment:
```suggestion
        flume_devices = await hass.async_add_executor_job(
                FlumeDeviceList,
                username,
                password,
                client_id,
                client_secret,
                flume_token_full_path,
        )
```

Reply:
Thank you.  I missed converting it to FlumeAuth!

Reply:
Fixed